intro-pkgs.Rmd

```diff
@@ -113,5 +113,8 @@ Now that you have an idea of the tools we can use for HTTP testing, we'll now cr
 
 Our minimal package will use httr. However, it will help you understand concepts even if you end up using crul or curl.[^limits]
 
+Test doubles such as mocks and fakes are helpful tools, but they also introduce their own complexity.
```

Comment:
"programs design" is a bit unclear.

Reply:
I think the mention of the chapter should best happen in the conclusion as it might be confusing and overwhelming in the intro.

Reply:
I know I said otherwise earlier. :thinking: Maybe the paragraph should be much shorter and not mention code design / code smell as it makes it look potentially scary right from the intro.

Reply:
I think there's a very real risk here to overbuild and overcomplicate the testing, while missing deeper flaws.
So I think a warning is warranted.

Reply:
fair enough!